src/App.tsx

```diff
@@ -1,10 +1,64 @@
+/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useEffect, useState } from 'react';
+import { UserWarning } from './components/UserWarning';
+import { getTodos, USER_ID } from './api/todos';
+
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotifications';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [valueError, setValueError] = useState('');
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>('all');
+  const [isNotificationHidden, setIsNotificationHidden] = useState(true);
+
+  const autoHideNotification = () => {
+    setTimeout(() => {
+      setIsNotificationHidden(true);
+    }, 3000);
+  };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const todosFromServer = await getTodos();
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setValueError('Unable to load todos');
+        setIsNotificationHidden(false);
+        autoHideNotification();
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const filteredTodos = todos.filter(todo => {
+    if (filter === 'active') {
+      return !todo.completed;
+    }
+
+    if (filter === 'completed') {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
+  const handleFilter = (selectedFilter: 'all' | 'active' | 'completed') => {
+    setFilter(selectedFilter);
+  };
```

Comment:
Its could looking more presentable, if you create enum with types of filters and filtered todos with switch case

Reply:
Good point, fix it